crates/sui-core/src/consensus_adapter.rs

```diff
@@ -440,41 +440,33 @@ impl ConsensusAdapter {
         &self,
         positions: Vec<AuthorityName>,
     ) -> (usize, bool) {
-        if self.authority_is_low_scoring(&self.authority) {
+        let low_scoring_authorities = self.low_scoring_authorities.load().load_full();
```

Comment:
@mwtian this is actually a double-nested struct:
```
low_scoring_authorities: ArcSwap<Arc<ArcSwap<HashMap<AuthorityName, u64>>>>
```
I'm not sure, but it doesn't seem reasonable, I'd expect just a single `ArcSwap` to be enough. @akichidis and @laura-makdah have more context on that. Let's land the fix and we can see whether we should flatten the struct?

Reply:
Sounds good.

Reply:
The inner arcswap is used by the consensus handler to update the values, and the outer arcswap is used at the reconfiguration time to supply a new shared map because the consensus handler gets recreated on every epoch the node is a validator, but the consensus adapter doesn't get shutdown or recreated. I explored shutting down and recreating the consensus adapter to avoid the outer arcswap, but it was too big and risky a change to make along with the scoring PR. I also explored making this a flattened struct and that resulted in the few days we observed the map wasn't getting updated.

Reply:
Thanks @laura-makdah for the context!